src/model/version.entity.ts

```diff
@@ -0,0 +1,23 @@
+import { Entity, PrimaryKey, Property, ManyToOne } from '@mikro-orm/core';
+import { Package } from './package.entity';
+
+@Entity()
+export class Version {
+  @PrimaryKey()
+  id!: number;
```

Comment:
Make primary key as composite key <package, version> - it will be simpler and better index. 
Is probably will work if you put @PrimaryKey() on `id!` and `version!` properties.